packages/data-context/src/sources/migration/codegen.ts

```diff
@@ -340,91 +340,86 @@ export function renameSupportFilePath (relative: string) {
-      const isDefaultCT = key === 'component' && specPattern === ext
+        if (!value) {
+          return acc
+        }
+
+        const { testFiles, ignoreTestFiles, ...rest } = value
+
+        // don't include if it's the default! No need.
+        const specPattern = getSpecPattern(cfg, key)
+        const ext = '**/*.cy.{js,jsx,ts,tsx}'
+        const isDefaultE2E = key === 'e2e' && specPattern === `cypress/e2e/${ext}`
+        const isDefaultCT = key === 'component' && specPattern === ext
+
+        if (isDefaultE2E || isDefaultCT) {
+          return {
+            ...acc, [key]: {
+              ...rest,
+              ...acc[key],
+            },
+          }
+        }
 
-      if (isDefaultE2E || isDefaultCT) {
         return {
           ...acc, [key]: {
             ...rest,
             ...acc[key],
+            specPattern,
           },
         }
       }
-
-      return {
-        ...acc, [key]: {
-          ...rest,
-          ...acc[key],
-          specPattern,
-        },
-      }
-    }
-
-    if (key === 'integrationFolder') {
-      return {
-        ...acc,
-        e2e: { ...acc.e2e, specPattern: getSpecPattern(cfg, 'e2e') },
-      }
-    }
-
-    if (key === 'componentFolder') {
-      return {
-        ...acc,
-        component: { ...acc.component, specPattern: getSpecPattern(cfg, 'component') },
-      }
-    }
-
-    if (key === 'testFiles') {
-      return {
-        ...acc,
-        e2e: { ...acc.e2e, specPattern: getSpecPattern(cfg, 'e2e') },
-        component: { ...acc.component, specPattern: getSpecPattern(cfg, 'component') },
-      }
-    }
-
-    if (key === 'ignoreTestFiles') {
-      return {
-        ...acc,
-        e2e: { ...acc.e2e, specExcludePattern: val },
-        component: { ...acc.component, specExcludePattern: val },
-      }
-    }
-
-    if (key === 'supportFile') {
-      return {
-        ...acc,
-        e2e: { ...acc.e2e, supportFile: val },
-      }
-    }
-
-    if (key === 'baseUrl') {
-      return {
-        ...acc,
-        e2e: { ...acc.e2e, [key]: val },
-      }
+      case 'integrationFolder':
+        return {
+          ...acc,
+          e2e: { ...acc.e2e, specPattern: getSpecPattern(cfg, 'e2e') },
+        }
+      case 'componentFolder':
+        return {
+          ...acc,
+          component: { ...acc.component, specPattern: getSpecPattern(cfg, 'component') },
+        }
+      case 'testFiles':
+        return {
+          ...acc,
+          e2e: { ...acc.e2e, specPattern: getSpecPattern(cfg, 'e2e') },
+          component: { ...acc.component, specPattern: getSpecPattern(cfg, 'component') },
+        }
+      case 'ignoreTestFiles':
+        return {
+          ...acc,
+          e2e: { ...acc.e2e, specExcludePattern: val },
+          component: { ...acc.component, specExcludePattern: val },
+        }
+      case 'supportFile':
+        return {
+          ...acc,
+          e2e: { ...acc.e2e, supportFile: val },
```

Comment:
hmm sorry @BlueWinds...sanity check though, currently in 9x the support file is shared between e2e and component right? Shouldn't this map to both?

Reply:
Hm, it does sound like it should to me. @lmiller1990, thoughts?

Reply:
I asked exactly the same question at the time - Brian said nope, we just map to `e2e.supportFile`. During CT setup we will create a new file called `cypress/support/component.js`.

(Yes - this means that CT users will have some small inconvenience during migration from 9.x -> 10.x. But CT users are in the minority).

You can ping in the migration channel if you feel this is weird and should be revisited.

Reply:
@lmiller1990 I did ask there as well. No response yet.